listenbrainz/tests/integration/test_api.py

```diff
@@ -383,6 +383,24 @@ def test_too_large_listen(self):
         self.assert400(response)
         self.assertEqual(response.json['code'], 400)
 
+    def test_empty_track_name(self):
+        """ Test for invalid submission in which a listen contains an empty track name
+        """
+        with open(self.path_to_data_file('empty_track_name.json'), 'r') as f:
+            payload = json.load(f)
+        response = self.send_data(payload)
+        self.assert400(response)
+        self.assertEqual(response.json['code'], 400)
+
+    def test_empty_artist_name(self):
+        """ Test for invalid submission in which a listen contains an empty track name
+        """
+        with open(self.path_to_data_file('empty_artist_name.json'), 'r') as f:
+            payload = json.load(f)
+        response = self.send_data(payload)
+        self.assert400(response)
+        self.assertEqual(response.json['code'], 400)
+
```

Comment:
it'd be good to also include tests for missing artist_name and track_name fields. We could use the same json files and just `del` the field before sending it.

I had a quick look and It seems like there are no tests anywhere for `validate_listen` (other than these two new ones), which seems like an oversight. We could add some more to a new `listenbrainz/webserver/views/test/test_api_tools.py`, which just calls this method without having to pass a listen through the whole webserver.

Reply:
I'll do the former here, but lets open a ticket for adding better tests and get this PR out of the way.